common/config/cluster.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"go.uber.org/multierr"
+)
+
+type (
+	// ClusterMetadata contains the all clusters participated in a replication group(aka XDC/GlobalDomain)
+	ClusterMetadata struct {
+		EnableGlobalDomain bool `yaml:"enableGlobalDomain"`
+		// FailoverVersionIncrement is the increment of each cluster version when failover happens
+		// It decides the maximum number clusters in this replication groups
+		FailoverVersionIncrement int64 `yaml:"failoverVersionIncrement"`
+		// PrimaryClusterName is the primary cluster name, only the primary cluster can register / update domain
+		// all clusters can do domain failover
+		PrimaryClusterName string `yaml:"primaryClusterName"`
+		// MasterClusterName is deprecated. Please use PrimaryClusterName.
+		MasterClusterName string `yaml:"masterClusterName"`
+		// CurrentClusterName is the name of the cluster of current deployment
+		CurrentClusterName string `yaml:"currentClusterName"`
+		// ClusterInformation contains information for each cluster within the replication group
+		// Key is the clusterName
+		ClusterInformation map[string]ClusterInformation `yaml:"clusterInformation"`
```

Comment:
Can we change this to "ReplicationGroup"? 

e.g. `ReplicationGroup map[string]ClusterInformation`

Reply:
Yeah, having the same name for both map name and item within that map does not seem right.

`ClusterGroup` seems reasonable. Or maybe just simply `Clusters`?
I think it would also make sense to rename `ClusterMetadata` to (maybe) `ClusterGroupMetadata` as it contains config for multiple clusters not just single one.

Reply:
I think having "Group" explicitly is more friendly as the "s' is quite hard to read/recognize. 

Yes, I love the idea of `ClusterGroupMetadata`

Reply:
Renamed:
- `ClusterInformation` to `ClusterGroup`
- `ClusterMetadata` to `ClusterGroupMetadata`
This also means introducing new fields in yaml deprecating old ones. Otherwise having them different could lead to some confusion.